add external istiod initial documentation (#8403)
istio/istio.io: content/en/docs/setup/additional-setup/external-istiod/index.md

---
title: Installing External Istiod [experimental]
description: Install external Istiod and remote config cluster.
weight: 46
keywords: [external,istiod,remote,config]
aliases:
    - /docs/setup/kubernetes/additional-setup/external-istiod/
owner: istio/wg-environments-maintainers
test: no
---

{{< warning >}}
This feature is actively in [development](https://github.com/istio/community/blob/master/FEATURE-LIFECYCLE.md) and is
considered `pre-alpha`.
{{< /warning >}}

## Introduction

> The {{< gloss >}}external control plane{{< /gloss >}} deployment model enables mesh operators to install and manage mesh control planes on separate external clusters. This deployment model allows a clear separation between mesh operators and mesh admins. Istio mesh operators can now run Istio control planes for mesh admins while mesh admins can still control the configuration of the control plane without worrying about installing or managing the control plane.

Reviewer: ```suggestion
The {{< gloss >}}external control plane{{< /gloss >}} deployment model enables mesh operators to install and manage mesh control planes on separate external clusters. This deployment model allows a clear separation between mesh operators and mesh admins. Istio mesh operators can run Istio control planes for mesh admins while mesh admins control the configuration of the control plane without worrying about installing or managing it.
```

## Requirements

> ### Cluster

Reviewer: ```suggestion
### Clusters
```

This guide requires that you have two Kubernetes clusters with any of the

> supported Kubernetes versions: {{< supported_kubernetes_versions >}}. First cluster is the {{< gloss >}}external control plane{{< /gloss >}} cluster where it has Istio default profile installed in the <code>istio-system</code> namespace.   It also has external istiod installed in the <code>external-istiod</code> namespace.  The external istiod is exposed on the ingress gateway from the <code>istio-system</code> namespace.  The second cluster is a {{< gloss >}}remote cluster{{< /gloss >}} which also provides configuration for the external istiod.

Reviewer 2: It's clearer to show this with a picture

Reviewer: The purpose of the `external-istiod` namespace is unclear. I added my "best guess" below.
```suggestion
supported Kubernetes versions: {{< supported_kubernetes_versions >}}.

The first cluster is the {{< gloss >}}external control plane{{< /gloss >}} cluster with the Istio `default` profile installed in the `istio-system` namespace. 
An ingress gateway, which provides mesh sidecars access to the external istiod, is also installed on this cluster in the `external-istiod` namespace.

The second cluster is a {{< gloss >}}remote cluster{{< /gloss >}} hosting the mesh. Its Kubernetes API server also provides the configuration for the control plane (istiod) running in the external cluster.
```

Author: Thanks for the feedback!  I'll add a diagram for sure, once I get the tests automated.

The ingress gateway is installed in the istio-system namespace, from the istio default profile.  External-istiod namespace hosts the istiod in the first cluster.

Reviewer: That doesn't seem to make much sense, since the default profile already has istiod. Maybe once you add a diagram it will make more sense, but either way, I think this warrants more explanation. The conceptual model is simple: one istiod in the external cluster with one ingressgateway to access it. Why there are 2 namespaces with 2 istiod's is counter intuitive, so that needs to be explained, and ideally illustrated in a diagram.

Author: Agreed it is confusing.  Basically the default profile has istiod and ingress gw... the sole purpose of that istiod is to manage the ingress gw.

Then external istiod (2nd istiod) is exposed on the ingress gw.

Reviewer: So the actual control plane is the istiod installed in the `external-istiod` namespace. In that case, I would suggest putting more emphasis on it, and downplay the second Istio installed in the `istio-system'  namespace, since it's just an implementation detail. What profile is used for the `external-istiod` Istio, also `default`? You should mention that. The profile used for the other istiod is really not important.

Why is the Istio in the `istio-system` namespace using the `default` profile anyway. If its only purpose is to configure the ingress gateway, couldn't the minimal profile be used for it?

Author: Yes both are using default profiles.

min profile for istio-system ns could work however min profile doesn't contain the ingress gw.

Author: reworded a bit, hope it is clear now!

### API Server Access

The API Server in the remote config cluster must be accessible to the external control plane cluster. Many cloud providers make API Servers publicly accessible via network

> load balancers (NLB). If the API Server is not directly accessible, you will

Reviewer: ```suggestion
load balancers (NLBs). If the API Server is not directly accessible, you will
```

have to modify the installation procedure to enable access. For example, the
[east-west](https://en.wikipedia.org/wiki/East-west_traffic) gateway used in
the multi-network and primary-remote configurations could also be used
to enable access to the API Server.

## Environment Variables

> This guide will refer to two clusters named `external_cp_cluster` and `user_cluster`. The following environment variables will be used throughout to simplify the instructions:

Reviewer: I would suggest simpler names:
```suggestion
This guide will refer to two clusters named `external_cluster` and `remote_cluster`. The following environment variables will be used throughout to simplify the instructions:
```

Variable | Description
-------- | -----------

> `CTX_EXTERNAL_CP` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the external control plane cluster.
> `CTX_USER_CLUSTER` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the remote config cluster.
> `REMOTE_ISTIOD_ADDR` | The external istiod host name for `user_cluster` to access external istiod.
> `SSL_SECRET_NAME` | The secret name used by the `external-istiod-gw` gateway resource on the external control plane cluster.

Reviewer: How about:
```suggestion
`CTX_EXTERNAL_CLUSTER` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the external control plane cluster.
`CTX_REMOTE_CLUSTER` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the remote config cluster.
`EXTERNAL_ISTIOD_ADDR` | The external host name for the `remote_cluster` to access the external istiod.
`SSL_SECRET_NAME` | The secret name used to access the ingress gateway on the external control plane cluster.
```

For example:

{{< text bash >}}
$ export CTX_EXTERNAL_CP=external_cp_cluster
$ export CTX_USER_CLUSTER=user_cluster
$ export REMOTE_ISTIOD_ADDR=myexternal-istiod.cloud.com
$ export SSL_SECRET_NAME = myexternal-istiod-secret
{{< /text >}}

## Setup

### Setup the external control plane cluster

Create the Istio configuration for `external_cp_cluster`, using the default profile with the following ports on the ingress gateway to expose the external istiod:

{{< text bash >}}
$ cat <<EOF > external-cp.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
 namespace: istio-system
spec:
 components:
   ingressGateways:
     - name: istio-ingressgateway
       enabled: true
       k8s:
         service:
           ports:
             - port: 15021
               targetPort: 15021
               name: status-port
             - port: 15012
               targetPort: 15012
               name: tls-xds
             - port: 15017
               targetPort: 15017
               name: tls-webhook
EOF
{{< /text >}}

> Apply the configuration in `external_cp_cluster` in the <code>istio-system</code> namespace:

Reviewer: ```suggestion
Apply the configuration in `external_cp_cluster` in the `istio-system` namespace:
```

{{< text bash >}}
$ istioctl apply -f external-cp.yaml --context="${CTX_EXTERNAL_CP}"
{{< /text >}}

> Expose the **to be installed** external istiod on Istio ingress gateway installed in the <code>istio-system</code> namespace. Create the Istio network configuration:

Reviewer: ```suggestion
Create the Istio network configuration to expose the **yet to be installed** external istiod ingress gateway in the `istio-system` namespace. 
```

     hosts:
     - "$REMOTE_ISTIOD_ADDR"
---
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
   name: external-istiod-vs
   namespace: external-istiod
spec:
   hosts:
   - $REMOTE_ISTIOD_ADDR
   gateways:
   - external-istiod-gw
   http:
   - match:
     - port: 15012
     route:
     - destination:
         host: istiod.external-istiod.svc.cluster.local
         port:
           number: 15012
   - match:
     - port: 15017
     route:
     - destination:
         host: istiod.external-istiod.svc.cluster.local
         port:
           number: 443
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
 name: external-istiod-dr
 namespace: external-istiod
spec:
 host: istiod.external-istiod.svc.cluster.local
 trafficPolicy:
   portLevelSettings:
   - port:
       number: 15012
     tls:
       mode: SIMPLE
     connectionPool:
       http:
         h2UpgradePolicy: UPGRADE
   - port:
       number: 443
     tls:
       mode: SIMPLE
EOF
{{< /text >}}

Apply the Istio configuration in `external_cp_cluster`:

{{< text bash >}}
$ kubectl apply -f external-istiod-gw.yaml --context="${CTX_EXTERNAL_CP}"
{{< /text >}}

### Setup remote cluster

> Generate the Istio configuration for `user_cluster` and the <code>external-istiod</code> namespace:

Reviewer: ```suggestion
Generate the Istio configuration for `user_cluster` and the `external-istiod` namespace:
```

{{< text bash >}}
$ cat <<EOF > remote-config-cluster.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
 namespace: external-istiod
spec:
 meshConfig:
   rootNamespace: external-istiod
   defaultConfig:
     discoveryAddress: $REMOTE_ISTIOD_ADDR:15012
     proxyMetadata:
       XDS_ROOT_CA: /etc/ssl/certs/ca-certificates.crt
       CA_ROOT_CA: /etc/ssl/certs/ca-certificates.crt
 components:
   pilot:
     enabled: false
   istiodRemote:
     enabled: true

 values:
   global:
     caAddress: $REMOTE_ISTIOD_ADDR:15012
     istioNamespace: external-istiod
   istiodRemote:
     injectionURL: https://$REMOTE_ISTIOD_ADDR:15017/inject
   base:
     validationURL: https://REMOTE_ISTIOD_ADDR:15017/validate
EOF
{{< /text >}}

Install the configuration in `user_cluster`:

{{< text bash >}}
$ istioctl apply -f remote-config-cluster.yaml --context="${CTX_USER_CLUSTER}"
{{< /text >}}

> You may notice the ingress gateway in `user_cluster` is not running yet.  This is normal as the ingress gateway won't reach running until its external Istiod reaches running, which you will install next.

Reviewer: ```suggestion
You may notice the ingress gateway in `user_cluster` is not running yet. This is expected until the external istiod is running, which you will install next.
```

{{< text bash >}}
$ cat <<EOF > external-istiod.yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
 namespace: external-istiod
spec:
 meshConfig:
   defaultConfig:
     discoveryAddress: $REMOTE_ISTIOD_ADDR:15012
     rootNamespace: external-istiod
     proxyMetadata:
       XDS_ROOT_CA: /etc/ssl/certs/ca-certificates.crt
       CA_ROOT_CA: /etc/ssl/certs/ca-certificates.crt
 components:
   base:
     enabled: false
   ingressGateways:
   - name: istio-ingressgateway
     enabled: false
 values:
   global:
     caAddress: $REMOTE_ISTIOD_ADDR:15012
     istioNamespace: external-istiod
     operatorManageWebhooks: true
   pilot:
     env:
       INJECTION_WEBHOOK_CONFIG_NAME: ""
       VALIDATION_WEBHOOK_CONFIG_NAME: ""
EOF
{{< /text >}}

Apply the Istio configuration in `external_cluster`:

{{< text bash >}}
$ istioctl apply -f external-istiod.yaml --context="${CTX_EXTERNAL_CLUSTER}"
{{< /text >}}

### Validate the installation

Confirm the Istio ingress gateway in `remote_cluster` is running.

{{< text bash >}}
$ kubectl get pod -l app=istio-ingressgateway -n external-istiod --context="${CTX_REMOTE_CLUSTER}"
{{< /text >}}

> Deploy the sleep sample in `remote_cluster` with a namespace has [automatic sidecar injection](/docs/setup/additional-setup/sidecar-injection/#automatic-sidecar-injection) enabled.  The sleep pod should reach running in a few seconds.

Reviewer: This isn't really much of a validation. Can't you at least run helloworld, and test that you can access it through the ingress gateway?

Author: This is the most critical testing we need :) because getting the sleep pod up running means we have sidecar injected from external istiod and able to establish the connection with external istiod. :)   

But I'd be happy to add a link to point to helloworld or something to test the gateway as well.

Author: addressed, I think!

---
title: Installing External Istiod [experimental]
description: Install external Istiod and remote config cluster.
weight: 46
keywords: [external,istiod,remote,config]
aliases:
    - /docs/setup/kubernetes/additional-setup/external-istiod/
owner: istio/wg-environments-maintainers
test: no
---

{{< warning >}}
This feature is actively in [development](https://github.com/istio/community/blob/master/FEATURE-LIFECYCLE.md) and is
considered `pre-alpha`.
{{< /warning >}}

## Introduction

> The {{< gloss >}}external control plane{{< /gloss >}} deployment model enables mesh operators to install and manage mesh control planes on separate external clusters. This deployment model allows a clear separation between mesh operators and mesh admins. Istio mesh operators can run Istio control planes for mesh admins while mesh admins control the configuration of the control plane without worrying about installing or managing it.

Reviewer 3: link to the deployment models guide.

Author: Just added!

> The {{< gloss >}}external control plane{{< /gloss >}} deployment model enables mesh operators to install and manage mesh control planes on separate external clusters. This deployment model allows a clear separation between mesh operators and mesh admins. Istio mesh operators can run Istio control planes for mesh admins while mesh admins control the configuration of the control plane without worrying about installing or managing it.

Reviewer 3: I think the wording of the first sentence should be something like:

```
This guide walks you through the installation of an external control plane. In
this deployment model, the mesh operator configures the mesh to use a
control plane residing outside the mesh. An external control plane can reside in
another cluster or be managed by a cloud vendor.
```

Reviewer 3: Also, please confirm that we have this content somewhere in our deployment models guide.

Author: Thanks!  I 've added the first line you had. I think the other messages are similar to what I have.  Yes this is also covered in deployment guide like 
```A cloud vendor’s managed control plane is a typical example of an external control plane.``` 
```Instead of running control planes in primary clusters inside the mesh, a service mesh composed entirely of remote clusters can be controlled by an external control plane. This provides isolated management and complete separation of the control plane deployment from the data plane services that comprise the mesh.```

> The {{< gloss >}}external control plane{{< /gloss >}} [deployment model](/docs/ops/deployment/deployment-models/#control-plane-models) enables mesh operators to install and manage mesh control planes on separate external clusters. This deployment model allows a clear separation between mesh operators and mesh admins. Istio mesh operators can run Istio control planes for mesh admins while mesh admins control the configuration of the control plane without worrying about installing or managing it.

Reviewer 3: Please see my previous comment regarding the re-wording of this paragraph. Also, I think we probably want to be more clear about the fact that the remote cluster is serving 2 distinct purposes:

- API Server used to store the configuration for the external control plane
- Runs user workloads (i.e. the mesh)

Author: I think this is explained in the #clusters section:

```
The second cluster is a {{< gloss >}}remote cluster{{< /gloss >}} hosting the mesh. Its Kubernetes API server also provides 
the configuration for the control plane (istiod) running in the external cluster.
```

## Requirements

### Clusters

This guide requires that you have two Kubernetes clusters with any of the
supported Kubernetes versions: {{< supported_kubernetes_versions >}}.

> The first cluster is the {{< gloss >}}external control plane{{< /gloss >}} cluster with the Istio `default` profile installed in the `external-istiod` namespace. This istiod serves as the the external control plane for the second cluster.  An ingress gateway is installed in the `istio-system` namespace. The ingress gateway provides mesh sidecars access to the external istiod in the `external-istiod` namespace.

Reviewer 3: `the the`

Author: updated

> The first cluster is the {{< gloss >}}external control plane{{< /gloss >}} cluster with the Istio `default` profile installed in the `external-istiod` namespace. This istiod serves as the the external control plane for the second cluster.  An ingress gateway is installed in the `istio-system` namespace. The ingress gateway provides mesh sidecars access to the external istiod in the `external-istiod` namespace.

Reviewer 3: Should be `the first cluster *contains* the external control plane`...

Reviewer 3: No need to mention profiles here, it's too much detail for an introduction.

Reviewer 3: Try to use `control plane` instead of `istiod` throughout. Istiod is just the name of the process and shouldn't be mentioned unless it's explicitly needed somewhere in the doc.

Author: updated

The second cluster is a {{< gloss >}}remote cluster{{< /gloss >}} hosting the mesh. Its Kubernetes API server also provides the configuration for the control plane (istiod) running in the external cluster.

> 

Reviewer 3: Please add a diagram

Author: Yes, will work on this in a follow up PR after able to add tests (see https://github.com/istio/istio.io/pull/8431)

Reviewer 3: why not just include it now? It will be easier than having to go through another PR process, no?

### API Server Access

The API Server in the remote config cluster must be accessible to the external control plane cluster. Many cloud providers make API Servers publicly accessible via network
load balancers (NLBs). If the API Server is not directly accessible, you will
have to modify the installation procedure to enable access. For example, the
[east-west](https://en.wikipedia.org/wiki/East-west_traffic) gateway used in
the multi-network and primary-remote configurations could also be used
to enable access to the API Server.

## Environment Variables

This guide will refer to two clusters named `external_cluster` and `remote_cluster`. The following environment variables will be used throughout to simplify the instructions:

Variable | Description
-------- | -----------

> `CTX_EXTERNAL_CLUSTER` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the external control plane cluster.

Reviewer 3: Let's just use these for brevity: `CTX_EXTERNAL`, `CTX_REMOTE`, `ISTIOD_ADDR`, `SECRET_NAME`

Author: Can we leave out these changes till we have more time?  Trying to get this PR in first so we can claim external istiod alpha before the release.

Reviewer 3: I know this is pre-alpha, but readability is still important. I think trimming these down should be a relatively simple change, no?

> `CTX_REMOTE_CLUSTER` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the remote config cluster.

Reviewer 3: Just say `remote cluster` instead of `remote config cluster`.

Author: updated

> `EXTERNAL_ISTIOD_ADDR` | The external host name for the `remote_cluster` to access the external istiod.

Reviewer 3: The wording in this description is a little unclear. Maybe something like `The host name that the remote cluster will use to access the external control plane`.

Author: Updated

`SSL_SECRET_NAME` | The secret name used to access the ingress gateway on the external control plane cluster.

For example:

{{< text bash >}}
$ export CTX_EXTERNAL_CLUSTER=external_cluster
$ export CTX_REMOTE_CLUSTER=remote_cluster
$ export REMOTE_ISTIOD_ADDR=myexternal-istiod.cloud.com
$ export SSL_SECRET_NAME = myexternal-istiod-secret
{{< /text >}}

## Setup

> ### Setup the external control plane cluster

Reviewer 3: This is not the first step, IIUC.  The installation should be:

1. Install the config cluster (in cluster2)
1. Install the primary cluster (in cluster1) that points to the config cluster
1. Install the remote cluster (in cluster2)

Author: The steps I have is the simplest steps to set up things.

If you set up config cluster first, you don't know the remote pilot addr yet then you have to go back to re-configure the config cluster.

That is why we setup the first cluster to get the gw and istiod exposed first before anything else.

Also, in external istiod, the config cluster has to be set up first before the external istiod is up.  This is different from primary-remote.

Reviewer 3: You shouldn't need a remote pilot address to set up a config cluster. A config cluster should effectively be it's own profile that just installs CRDs and maybe some secrets ... nothing else.

This flow as-described here is confusing because it's a chicken-or-the-egg situation. It also violates the separation of the mesh operator and admin roles as you described above. You're effectively setting up a remote before you install the control plane.

This needs to be fixed soon ... definitely before 1.9.

Create the Istio configuration for `external_cluster`, using the default profile with the following ports on the ingress gateway to expose the external istiod:

{{< text bash >}}

> $ cat <<EOF > external-cp.yaml

Reviewer 3: Should we use ingress-gateway or east-west gateway?

@stevenctl @costinm any thoughts?

Reviewer 4: I would say eastwest – the whole idea was that we'd have different gateways to allow hardening and scaling the gateway used for non-ingress purposes separately from how we scale the gateway responsible for external traffic coming in.

Author: Note: this is the istio + ingress gateway that istiod is exposed on.  It is not just the gateway.  And this is slight different than the east-west gw we used in multicluster to allow communications among service a in cluster 1 to reach to service b in cluster 2. It is for sidecar in remote cluster to reach to external istiod.

Reviewer 4: We use eastwest to reach istiod in the primary remote case. Shouldn't this be similar?

Reviewer: I suppose neither `ingress-gateway` nor `eastwest-gateway` are very good names for this one. Maybe just call it `controlplane-gateway`?

Author: Thanks Frank, updated!

apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
 namespace: istio-system
spec:
 components:
   ingressGateways:
     - name: istio-ingressgateway
       enabled: true
       k8s:
         service:
           ports:
             - port: 15021
               targetPort: 15021
               name: status-port
             - port: 15012
               targetPort: 15012
               name: tls-xds
             - port: 15017
               targetPort: 15017
               name: tls-webhook
EOF
{{< /text >}}

Apply the configuration in `external_cluster` in the `istio-system` namespace:

{{< text bash >}}
$ istioctl apply -f external-cp.yaml --context="${CTX_EXTERNAL_CLUSTER}"
{{< /text >}}

> Create the Istio network configuration to expose the **yet to be installed** external istiod on the ingress gateway in the `istio-system` namespace:

Reviewer 3: DIdn't you already install istiod in the previous step?

Author: The first istiod is just to manage the ingress gw.  It was not an external istiod.

Reviewer 3: That is really confusing. We can't let this go to alpha with this flow in place.

{{< text bash >}}

> $ cat <<EOF > external-istiod-gw.yaml

Reviewer 3: We should make this part of the `multicluster` samples to provide a gateway for istiod that does TLS termination

Author: It currently is using termination.  however, for automated testing, I'm going to use auto passthrough due to limitation of our test framework without trust-able host/key/certs.

Reviewer 3: Can you raise an issue for our test framework so we can get this addressed for 1.9?

> ---

Reviewer 3: The path on this doc should not include `istiod`, it should be `.../external-control-plane`.

Author: Updated

> title: Installing External Istiod [experimental]

Reviewer 3: Please replace `istiod` with `control plane` throughout this doc.

> description: Install external Istiod and remote config cluster.

Reviewer 3: Just drop `config cluster` from the description for now ... we don't really document that yet.

Author: Done

weight: 46
keywords: [external,istiod,remote,config]
aliases:

>     - /docs/setup/kubernetes/additional-setup/external-istiod/

Reviewer 3: why do you need an alias? My understanding was that aliases were mainly for redirects to old pages that were moved/replaced?

Author: probably not needed... removed.

test: no
---

{{< warning >}}
This feature is actively in [development](https://github.com/istio/community/blob/master/FEATURE-LIFECYCLE.md) and is
considered `pre-alpha`.
{{< /warning >}}

## Introduction

This guide walks you through the installation of an external control plane. The {{< gloss >}}external control plane{{< /gloss >}} [deployment model](/docs/ops/deployment/deployment-models/#control-plane-models) enables mesh operators to install and manage mesh control planes on separate external clusters. This deployment model allows a clear separation between mesh operators and mesh admins. Istio mesh operators can run Istio control planes for mesh admins while mesh admins control the configuration of the control plane without worrying about installing or managing it.

## Requirements

### Clusters

This guide requires that you have two Kubernetes clusters with any of the
supported Kubernetes versions: {{< supported_kubernetes_versions >}}.

The first cluster contains the {{< gloss >}}external control plane{{< /gloss >}} installed
in the `external-istiod` namespace. This Istio control plane serves as the external control plane for the second cluster.

> An ingress gateway is installed in the `istio-system` namespace. The ingress gateway provides mesh sidecars access to the

Reviewer 3: I'd like more information here about how the TLS is working e2e

Author: So basically, the sidecars in remote cluster have key/certs to talk to gw when gw has trust-able host with trust-able key and certs.   The connection is terminated at the gateway, then upgraded to simple TLS from gateway to istiod.  Istiod uses JWT to validate the sidecars.  This is not perfect and there is a WI to improve this:  https://github.com/istio/istio/issues/28510  cc @howardjohn @stevenctl

to enable access to the API Server.

## Environment Variables

This guide will refer to two clusters named `external_cluster` and `remote_cluster`. The following environment variables will be used throughout to simplify the instructions:

Variable | Description
-------- | -----------
`CTX_EXTERNAL_CLUSTER` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the external control plane cluster.
`CTX_REMOTE_CLUSTER` | The context name in the default [Kubernetes configuration file](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) used for accessing the remote cluster.
`EXTERNAL_ISTIOD_ADDR` | The host name for the `remote_cluster` to access the external control plane.
`SSL_SECRET_NAME` | The secret name used to access the ingress gateway on the external control plane cluster.

For example:

{{< text bash >}}
$ export CTX_EXTERNAL_CLUSTER=external_cluster
$ export CTX_REMOTE_CLUSTER=remote_cluster
$ export REMOTE_ISTIOD_ADDR=myexternal-istiod.cloud.com

> $ export SSL_SECRET_NAME=myexternal-istiod-secret

Reviewer 2: Can not remember correctly about sds flow, should this secret be mounted to gateway in order to serve sds?

Author: This secret is specified later on in the gateway resource configuration, and the gateway will attempt to load the secret dynamically.
